core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java

```diff
@@ -72,7 +72,7 @@ public void resetMetaMasterConfigClient(MasterClientContext context) {
   }
 
   @Override
-  public synchronized void heartbeat() {
+  public void heartbeat() {
```

Comment:
I'm not entirely sure we are allowed to remove this `synchronized`. However, I know we can remove the one with `close()`, since it just calls another method which is already `synchronized`. If we just remove the `synchronized` on the `ConfigHashSync#close()`, that should remove the deadlock cycle, right?

I'm just concerned that `synchronized` might be required for `heartbeat()`.

Reply:
We have one thread that should be running heartbeat so there or not, it shouldn't really make a difference. I'll put it back.